src/library/simplified/extension/number-of-guests.js

```diff
@@ -128,6 +128,10 @@
     {
         b['numberOfGuests'] = b['entity.cruise.traveler.totalNumberOfTravelers'];
     }
+	else if (utag.isCruisePymt() && b["entity.checkout.cruise.passengerCount"])
+    {
+        b['numberOfGuests'] = b["entity.checkout.cruise.passengerCount"];
```

Comment:
why at the payment page the numberOfGuest will be different from the CKO page logic below?